pkg/tls/certificatemanagement/assets.go

```diff
@@ -0,0 +1,90 @@
+package certificatemanagement
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/dns"
+)
+
+const (
+	// PrometheusServerTLSSecretName is the TLS secret used when the authentication proxy container in tigera-prometheus
+	// is acting as a server and handles incoming requests.
+	PrometheusServerTLSSecretName = "tigera-prometheus-server-tls"
+)
+
+// TLSAsset should be instantiated at startup only, none of its members are exposed as they should be immutable.
+type TLSAsset struct {
+	secretName         string
+	serviceaccountName string
+	namespace          string
+	permittedDNSNames  []string
+	keyExtUsages       []x509.ExtKeyUsage
+}
+
+func (t TLSAsset) SecretName() string {
+	return t.secretName
+}
+
+func (t TLSAsset) DNSNames() []string {
+	return t.permittedDNSNames
+}
+
+func (t TLSAsset) ExtKeyUsages() []x509.ExtKeyUsage {
+	return t.keyExtUsages
+}
+
+// TLSAssetMap can look up all assets associated with a serviceaccount and determine if a CSR is requesting allowed
+// certificate contents.
+type TLSAssetMap map[string]TLSAsset
+
+var (
+	// TLSAssets Stores the secret name as key and asset as value.
+	TLSAssets           = make(TLSAssetMap)
+	PrometheusServerTLS TLSAsset
```

Comment:
The idea is that all of our TLS assets are defined here and initiated at startup.

Then the signature of `certificateManager.GetOrCreateKeyPair(...)` can be modified to get simply the TLSAsset as input.